libasynql/src/poggit/libasynql/base/SqlThreadPool.php

```diff
@@ -38,22 +43,41 @@ class SqlThreadPool implements SqlThread{
 	/** @var QueryRecvQueue */
 	private $bufferRecv;
 
+	/** @var DataConnector|null */
+	private $dataConnector = null;
+
+	/**
+	 * @param DataConnector $dataConnector
+	 */
+	public function setDataConnector(DataConnector $dataConnector): void {
+		$this->dataConnector = $dataConnector;
+	}
+
 	/**
 	 * SqlThreadPool constructor.
 	 *
 	 * @param callable $workerFactory create a child worker: <code>function(?Threaded $bufferSend = null, ?Threaded $bufferRecv = null) : {@link BaseSqlThread}{}</code>
 	 * @param int      $workerLimit   the maximum number of workers to create. Workers are created lazily.
 	 */
 	public function __construct(callable $workerFactory, int $workerLimit){
+		$this->notifier = new SleeperNotifier();
+		Server::getInstance()->getTickSleeper()->addNotifier($this->notifier, function() : void{
+			assert($this->dataConnector instanceof DataConnector); // otherwise, wtf
+			$this->dataConnector->checkResults();
+		});
+
 		$this->workerFactory = $workerFactory;
 		$this->workerLimit = $workerLimit;
 		$this->bufferSend = new QuerySendQueue();
 		$this->bufferRecv = new QueryRecvQueue();
-		$this->addWorker();
+
+		for($i = 0; $i < $workerLimit; $i++){
+			$this->addWorker();
```

Comment:
Do we need to maximize the number of workers directly upon startup? What is the problem with lazy worker initialization?